### ISSUE TITLE
Property injection not functional

### ISSUE THREAD

**Issue by chiefengineer** [simpleinjector/SimpleInjector#461]
Before anyone goes off a cliff preparing for a rant, yes, I read the documentation about how this is a Bad Idea. I get that. I also don't care right now as I am trying to work through an example as provided in a book (Adam Freeman's Pro ASP.NET MVC 5) that specifically uses property injection to demonstrate a point. 

I'd like to see it working. The real fun part is that I have tried this with both SimpleInjector and Ninject (started with Ninject as that is what the book is using but knowing that Ninject is getting a little long in the tooth, I decided to try SI). I have followed the SI documentation and the recommendations (this example project isn't rocket science so that ought to work) but both Ninject and SI fail the same way - the property isn't assigned when the concrete class is created. Both DI containers work just fine up to that point, it's just trying the property injection thing that isn't working. I'd would have thought that perhaps I was doing something incorrectly but given the wildly different syntax of the two, I would have expected a failure of one but not the other. Certainly never both at exactly the same point and with the same error. 

I have detailed things a bit more [extensively](https://stackoverflow.com/questions/46290135/property-injection-failing-with-both-ninject-and-simpleinjector)  if anyone wants to take a peek.

I would really appreciate some insight into what's going on (or not, for that matter) as I'd like to move to SI but I need to be able to trust it to Just Work and right now, I can't get that to happen.

Thanks!

**Comment by dotnetjunkie** [comment]
Cross post: https://stackoverflow.com/questions/46290135/property-injection-failing-with-both-ninject-and-simpleinjector.

Let's continue the discussion on Stackoverflow.

**Comment by chiefengineer** [comment]
Sure thing. I wanted to post here, specifically as I am new to SI and wanted to make sure I'm doing things correctly, but I wanted anyone with Ninject experience to have a crack at it as well as the same problem has manifested itself in both libraries and I have no idea why. Was hoping someone else had already encountered this and it would simply be a "correct my ignorance" answer.

**Comment by chiefengineer** [comment]
It appears that the simple web example *is* the simplest form of what is
showing the problem I have encountered. I translated the code to a console
app as you requested, however, I had to change so many things that the
result is nothing like what caused the problem (and, it does not display
the same behavior).
Should you wish to continue the investigation, I have included the relevant
portions of the code necessary to flesh out the basic Web site. Again, let
me stress that this isn't a full-blown website, it's an example from a book
and one near the beginning of the book to boot. It's small. One need only
start with an ASP.NET MVC "Empty" site after selecting both "MVC" and "Web
API" (check boxes) and then take the files in their respective directories
from the zip file and move them to the same directories in the new project.

I'm *really* curious as to why such a simple example can cause two wildly
different DI containers to fail in the same fashion without error.

Thanks!


"Bang the Button - It Just Works!"
